plugins/catalog-backend-module-gitlab/src/GitLabDiscoveryProcessor.test.ts

```diff
@@ -291,6 +344,47 @@ describe('GitlabDiscoveryProcessor', () => {
       expect(result).toHaveLength(1);
     });
 
+    it('can filter based on file existing', async () => {
+      const processor = getProcessor({ options: { checkFileExistence: true } });
+      setupFakeServer(GROUP_PROJECTS_URL, request => {
+        if (!request.include_subgroups) {
+          throw new Error('include_subgroups should be set');
+        }
+        switch (request.page) {
+          case 1:
+            return {
+              data: [
+                {
+                  id: 1,
+                  archived: false,
+                  default_branch: 'main',
+                  last_activity_at: '2021-08-05T11:03:05.774Z',
+                  web_url: 'https://gitlab.fake/1',
+                  path_with_namespace: '1',
+                },
+                {
+                  id: 1,
+                  archived: false,
+                  default_branch: 'main',
+                  last_activity_at: '2021-08-05T11:03:05.774Z',
+                  web_url: 'https://gitlab.fake/g/2',
+                  path_with_namespace: 'g/2',
+                },
+              ],
+            };
+          default:
+            throw new Error('Invalid request');
+        }
+      });
+
+      const result: any[] = [];
+      await processor.readLocation(GROUP_LOCATION_CUSTOM_BRANCH, false, e => {
+        result.push(e);
+      });
+      // If everything was set up correctly, we should have received the fake repo specified above
+      expect(result).toHaveLength(0);
```

Comment:
I think it's best to make sure this test both filters out one repo but includes another? Feeling the confidence that it adds is a bit low atm